### DISCUSSION TITLE
zod-to-openapi: Generating OpenAPI (Swagger docs) from Zod schemas

### DISCUSSION THREAD

**Discussion by georgyangelov** [colinhacks/zod#1107] [Show and tell]
Hey all! 👋 

First, thank you maintainers for building and maintaining Zod, it's awesome! We're starting to use it more and more in our company as the go-to validation solution for APIs, even teaching it in our internship program now.

In one of our projects we need to maintain an OpenAPI/Swagger documentation as well as the Zod definitions so until recently we've had to define each resource & input/output parameter twice. 

We made [zod-to-openapi](https://github.com/asteasolutions/zod-to-openapi) which generates OpenAPI definitions, routes and parameters based on Zod schemas while maintaining the ability to add custom OpenAPI-only metadata with a fluent interface.

We've also looked at other libraries which build OpenAPI definitions from Zod schemas automatically but found them lacking for our use-case in some areas, like the ability to have additional metadata (e.g. description) in one, the ability to generate path specs in other and ability to define which schema should be separately defined (and `$ref`erenced) and which one should be inlined. 

Our goal was for it to be as non-opinionated and flexible as possible, and with great ergonomics.

@colinhacks, It would be awesome if you could add [zod-to-openapi](https://github.com/asteasolutions/zod-to-openapi) to [the Ecosystem section](https://github.com/colinhacks/zod#ecosystem) in the readme. Completely fine if you don't, I just feel it's a useful tool that people like me would find more easily that way 😄

**Comment by scotttrinh** [comment]
@georgyangelov 

Feel free to open a PR against the README to add your project, looks great!

**Reply by georgyangelov** [reply]
Awesome, thank you! Here is the PR: https://github.com/colinhacks/zod/pull/1109
